src/gdb_main.c

```diff
@@ -115,18 +113,14 @@ static target_controller_s gdb_controller = {
 	.system = hostio_system,
 };
 
-int gdb_main_loop(target_controller_s *tc, bool in_syscall)
+/* execute gdb remote command stored in 'pbuf'. returns immediately, no busy waiting. */
+
+int gdb_main_loop(target_controller_s *tc, char *pbuf, size_t pbuf_size, size_t size, bool in_syscall)
 {
 	bool single_step = false;
 
 	/* GDB protocol main loop */
-	while (1) {
-		SET_IDLE_STATE(1);
-		size_t size = gdb_getpacket(pbuf, BUF_SIZE);
-		// If port closed and target detached, stay idle
-		if (pbuf[0] != '\x04' || cur_target) {
-			SET_IDLE_STATE(0);
-		}
+	{
```

Comment:
With the removal of the loop, we'd remove the comment as it's inaccurate and de-indent the entire block